src/core/thread/mle_router.cpp

```diff
@@ -1686,6 +1660,59 @@ void MleRouter::HandleTimeTick(void)
         }
 #endif
 
+        if (router.IsStateValid())
+        {
+            // Neighbor router age and link recovery
+            //
+            // If the device is an FTD child and has more than
+            // `mChildRouterLinks` neighbors, it uses a longer age,
+            // `kMaxNeighborAgeOnChild`, and removes the neighboring
+            // router upon expiration without trying to re-establish
+            // its link with it.
+            //
+            // Otherwise, if the device itself is a router, or it is an
+            // FTD child with `mChildRouterLinks` or fewer neighbors,
+            // it uses a shorter `kMaxNeighborAge`. Upon expiration, it
+            // tries to re-establish its link with the neighboring router.
+            // It sends a Link Request every second, up to `kMaxTxCount`.
+            // Each transmission is randomly delayed (one-second window).
+            // After the last attempt, we wait for the "Link Accept"
+            // timeout (~3 seconds) before removing the neighboring
+            // router.
+
+            if (IsChild())
+            {
+                uint16_t neighborCount = mRouterTable.GetNeighborCount(kLinkQuality1);
+
+                if ((neighborCount > mChildRouterLinks) && (age >= kMaxNeighborAgeOnChild))
+                {
+                    LogInfo("No Adv from router 0x%04x - have %u neighbor routers", router.GetRloc16(), neighborCount);
+                    RemoveNeighbor(router);
+                    continue;
+                }
+            }
+
+            if (age >= kMaxNeighborAge)
+            {
+                bool sendLinkRequest = true;
+
+                if (!router.IsWaitingForLinkAccept())
+                {
+                    LogInfo("No Adv from router 0x%04x - sending Link Request", router.GetRloc16());
+                }
+                else
+                {
+                    sendLinkRequest = (age < kMaxNeighborAge + kMaxTxCount * TimeMilli::kOneSecondInMsec);
```

Comment:
Should we avoid sending Link Request from Child whenever `neighborCount` > `mChildRouterLinks`?

Reply:
Yes. Good catch. Thanks. Will update this whole block.

Reply:
Updated in new push.